File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.py
def divisor(num):
    
    cnt = 0
    for i in range(1, int(num ** 0.5) + 1):
        if num % i == 0:
            cnt += 1
            if i ** 2 != num:
                cnt += 1
    return cnt

def solution(number, limit, power):
    
    ans = 0
    for n in range(1, number+1):
        div = divisor(n)
        
        if div <= limit:
            ans += div
        else:
            ans += power
    
    return ans

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
import java.util.*;
class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        
        // 에라토스테네스 체
        int[] divisor = new int[number + 1];
        for(int i = 1; i <= number; i++) {
            for(int j = i; j <= number; j+= i) {
                divisor[j]++;
            }
        }
        
        for(int i = 1; i < divisor.length; i++) {
            if (divisor[i] > limit) {
                answer += power;
            } else {
                answer += divisor[i];
            }
        }
        return answer;
    }
}